### DISCUSSION TITLE
Internal Server Error post matches

### DISCUSSION THREAD

**Discussion by thboss** [PhlexPlexico/G5API#79] [Help]
I get 500 error when posting this data (I'm using your web https://phlex.avidgamers.me)
```py
data = [
    {
        'user_id': user_id,
        'user_api': user_key,
        'server_id': 35,
        'team1_id': 181,
        'team2_id': 182,
        'title': '[PUG] Map {MAPNUMBER} of {MAXMAPS}',
        'is_pug': 1,
        'start_time': datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
        'ignore_server': 0,
        'max_maps': 1,
        'skip_veto': 1,
        'veto_first': 'team1'
    }
]
```

**Comment by PhlexPlexico** [comment]
I'm assuming you're using Python? The last error I see here is the following:

```
POST /matches 500 1008.292 ms - 67
::ffff:192.168.1.4 - - [06/Mar/2021:15:11:24 +0000] "POST /matches HTTP/1.1" 500 67 "-" "Python/3.7 aiohttp/3.6.2"
```

Is `user_key` also surrounded by quotes? Technically the API is sending back what is wrong with the request typically, so is there any output?
```js
catch (err) {
    res.status(500).json({ message: err.toString() });
  }
```

**Reply by thboss** [reply]
it works only when I set `ignore_server` to 1
also, can you add the maps list field? so I want to post the picked maps in discord directly on `/matches` instead of using vetoes endpoint, something like:
```py
data = [
    {
       other data ...,
       'maps': [
          'de_dust2',
          'de_mirage',
          'de_cache'
       ]
    }
]
```

**Reply by PhlexPlexico** [reply]
Please make sure you're checking the response. The reason this is failing is because your game server is not configured properly. There's an error being thrown is the following:

```
SyntaxError: Unexpected token U in JSON at position 0
    at JSON.parse (<anonymous>)
    at ServerRcon.isGet5Available (/home/phlex/G5API/utility/serverrcon.js:60:39)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /home/phlex/G5API/routes/matches/matches.js:714:10
POST /matches 500 913.066 ms - 67
```

Your game server is returning a bad value when checking `get5_web_available`, hence why `ignore_server` works, because you're ignoring the game server.

There was actually discussion of endpoints in vetoes before. I decided to do a veto endpoint specifically so that each time a map is vetoed, it goes to the endpoint. If you want to setup a match you can just provide one map in the map list.

**Reply by thboss** [reply]
I get an error here https://phlex.avidgamers.me/api/matches/151/config

**Reply by PhlexPlexico** [reply]
Uhh why do you have your team IDs set to 205 and 206? Those don't exist and probably why it's calling an error. I'll have to adjust the query but that's most definitely the reason.

**Reply by thboss** [reply]
I just fixed that by adding `flag` to create team body because of this https://github.com/PhlexPlexico/G5API/blob/f139f8ac230799d6326dabdf071a9a0b3aaeb21b/routes/matches/matches.js#L1123
I think must be a default value here?

**Reply by PhlexPlexico** [reply]
I don't think that's the case. It's failing on this kind 

https://github.com/PhlexPlexico/G5API/blob/f139f8ac230799d6326dabdf071a9a0b3aaeb21b/routes/matches/matches.js#L1113

Because team 205 does not exist, so it's trying to look at an undefined value and fails.

**Reply by PhlexPlexico** [reply]
For example, look at

https://phlex.avidgamers.me/api/teams/205
https://phlex.avidgamers.me/api/teams/206

Those teams don't exist, yet you have them set in your match config. This is why it's failing. If you put an actual team in the match information then it will work.
